examples/demo-nft-module/README.md

```diff
@@ -0,0 +1,481 @@
+
+2. Several state attributes. In this case it is an address of admin, and map of token id to owner address. Token id is u64 for simplicity.
+
+
+Context is a trait which allows the runtime to pass verified data to modules during execution. 
+Currently the only required method in Context sender(), which returns the address of the person who initiated the transaction (the signer).
+Context also inherits the Spec trait, which defines the concrete types used by the rollup for Hashing, persistent data Storage, digital Signatures, and Addresses. 
+The Spec trait allows rollups to easily tailor themselves to different ZK VMs. 
+By being generic over a Spec, a rollup can ensure that any potentially SNARK-unfriendly cryptography can be easily swapped out.
+
+
+# Implementing `sov_modules_api::Module` trait
+Before implementing `Module` trait, several things needs to be done:
+
+1. Add new dependencies: serde, borsh, sov-state
+2. Define 'native' feature flag to separate logic that is not needed in ZK context
+3. Define Call and Query messages
+4. Define Config
+
+## Preparation
+
+
+
+1. Define `native` feature in `Cargo.toml`:
+
+```toml
+[dependencies]
+anyhow = "1.0.62"
+borsh = { version = "0.10.3", features = ["bytes"] }
+serde = { version = "1", features = ["derive"] }
+serde_json = "1"
+
+sov-modules-api = { git = "https://github.com/Sovereign-Labs/sovereign.git", branch = "main", default-features = false }
+sov-modules-macros = { git = "https://github.com/Sovereign-Labs/sovereign.git", branch = "main" }
+sov-state = { git = "https://github.com/Sovereign-Labs/sovereign.git", branch = "main", default-features = false }
+
+
+[features]
+default = ["native"]
+serde = ["dep:serde", "dep:serde_json"]
+native = ["serde", "sov-state/native", "sov-modules-api/native"]
+```
+
+It is needed for optimizing module for execution in ZK context, where none of RPC related logic is needed.
+
+
+2. Define Call and Query messages. Call messages are used to change state of the module, and Query messages are used to read state of the module.
+
+```rust
+use sov_modules_api::Context;
+
+#[cfg_attr(feature = "native", derive(serde::Serialize), derive(serde::Deserialize))]
+#[derive(borsh::BorshDeserialize, borsh::BorshSerialize, Debug, PartialEq, Clone)]
+pub enum CallMessage<C: Context> {
+    Mint {
+        /// The id of new token. Caller is an owner
+        id: u64
+    },
+    Transfer {
+        /// The address to which the token will be transferred.
+        to: C::Address,
+        /// The token id to transfer
+        id: u64,
+    },
+    Burn {
+        id: u64,
+    }
+}
+
+
+/// This enumeration responsible for querying the nft module.
+#[derive(borsh::BorshDeserialize, borsh::BorshSerialize, Debug, PartialEq)]
+pub enum QueryMessage {
+    GetOwner { token_id: u64 },
+}
+```
+
+Query message is only used in native mode, because it does not change state in any way, since it is not used in state transition function.
```

Comment:
For clarity, how about? 


```suggestion
The query message is exclusively used in native mode as it does not alter the state. This is because it is not utilized in the state transition function.
```

Reply:
BTW, just curious, how do different modules read the state of other modules? Do they not use the query message?

Reply:
Module should expose public methods, using query messages adds overhead on serialization/deserialization.